lib/ocpp/v201/charge_point.cpp

```diff
@@ -4044,65 +4054,14 @@ void ChargePoint::handle_clear_display_message(const Call<ClearDisplayMessageReq
     this->message_dispatcher->dispatch_call_result(call_result);
 }
 
-void ChargePoint::handle_data_transfer_req(Call<DataTransferRequest> call) {
-    const auto msg = call.msg;
-    DataTransferResponse response;
-
-    if (this->callbacks.data_transfer_callback.has_value()) {
-        response = this->callbacks.data_transfer_callback.value()(call.msg);
-    } else {
-        response.status = DataTransferStatusEnum::UnknownVendorId;
-        EVLOG_warning << "Received a DataTransferRequest but no data transfer callback was registered";
-    }
-
-    ocpp::CallResult<DataTransferResponse> call_result(response, call.uniqueId);
-    this->message_dispatcher->dispatch_call_result(call_result);
-}
-
 std::optional<DataTransferResponse> ChargePoint::data_transfer_req(const CiString<255>& vendorId,
                                                                    const std::optional<CiString<50>>& messageId,
                                                                    const std::optional<json>& data) {
-    DataTransferRequest req;
-    req.vendorId = vendorId;
-    req.messageId = messageId;
-    req.data = data;
-
-    return this->data_transfer_req(req);
+    return this->data_transfer->data_transfer_req(vendorId, messageId, data);
```

Comment:
Is this function still necessary then?

Reply:
Yes since it doesnt handle a `DataTransfer.req` from the CSMS but sends a `DataTransfer.req` to the CSMS